src/main/kotlin/systems/ajax/malov/stockanalyzer/service/impl/FinnhubStockRecordClientApi.kt

```diff
@@ -15,22 +18,26 @@ class FinnhubStockRecordClientApi(
     @Value("\${api.finnhub.symbols}") private val symbols: List<String>,
 ) : StockRecordClientApi {
 
-    override fun getAllStockRecords(): List<MongoStockRecord> {
+    override fun getAllStockRecords(): Flux<MongoStockRecord> {
         val retrievalDate = Instant.now()
 
-        return symbols.mapNotNull {
-            retrieveStockRecord(it, retrievalDate)
-        }.toList()
+        return Flux.fromIterable(symbols)
+            .publishOn(Schedulers.boundedElastic())
+            .flatMap { symbol ->
```

Comment:
do you need flatMap here if logic is running on boundedElastic threads?

Reply:
We discussed this